### PR TITLE
PP-12758 Test docker scout

### PR DESCRIPTION
**Pull request by kbottla**
## WHAT
- Added test to check docker scout installation

### CODE REVIEW

**Comment by alphagov-pay-ci-concourse** [comment]
**No YAML differences detected between PR and master pkl files in Concourse Team pay-dev**

**Comment by alphagov-pay-ci-concourse** [comment]
# Changes for pay-deploy/concourse-runner.pkl

<details><summary>Diff of YAML generated from pay-deploy/concourse-runner.pkl</summary>

```diff
--- pay-ci/ci/pkl-pipelines/pay-deploy/concourse-runner.yml
+++ pkl-pipeline-pr/ci/pkl-pipelines/pay-deploy/concourse-runner.yml
@@ -175,6 +175,7 @@
     paths:
     - ci/docker/concourse-runner/**
     - ci/pipelines/concourse-runner.yml
+    - ci/scripts/test-concourse-runner.sh
   icon: github
 - name: adminusers-master
   type: git
```
</details>


<details><summary>Concourse set-pipeline dry-run for `pay-deploy/concourse-runner.pkl`</summary>

```diff
resources:
  resource concourse-runner-pr has changed:
  icon: github
  name: concourse-runner-pr
  source:
    access_token: ((github-access-token))
    disable_forks: true
    paths:
    - ci/docker/concourse-runner/**
    - ci/pipelines/concourse-runner.yml
+   - ci/scripts/test-concourse-runner.sh
    repository: alphagov/pay-ci
  type: pull-request
  
pipeline name: concourse-runner

Dry-run mode was set, exiting.
```

**Comment by alphagov-pay-ci-concourse** [comment]
The following files had no diff in the YAML between the main branch and this PR:

```
pay-deploy/deploy-to-production.yml
pay-deploy/deploy-to-staging.yml
pay-deploy/detect-secrets.yml
pay-deploy/infra-drift-detector.yml
pay-deploy/init-lock-pools.yml
pay-deploy/pact-broker.yml
pay-deploy/pay-js.yml
pay-deploy/pkl-pipeline-changes.yml
pay-deploy/pool-resource.yml
pay-deploy/prometheus-pushgateway.yml
```

File: ci/scripts/test-concourse-runner.sh [hunk 1]
```diff
@@ -23,6 +23,17 @@ https://maven.apache.org/xsd/settings-1.0.0.xsd">
 EOF
 
 ## Test docker by running app tests
-mvn --global-settings settings.xml clean verify
+mvn --global-settings settings.xml clean verify --no-transfer-progress
+
+## Test docker scout
+echo "Checking docker scout installation..."
+DOCKER_SCOUT_VERSION=$(docker scout version | grep "darwin/amd64")
```
Comment (jfharden) [inline ci/scripts/test-concourse-runner.sh:30]: This won't be the same version when it's installed by concourse, darwin is the macos platform. 

You can probably make this whole test just

```
if docker scout version >>/dev/null 2>&1; then
  echo "Docker scout is installed"
else
  ...
```

Comment (kbottla) [reply ci/scripts/test-concourse-runner.sh:30]: Thanks! updated

**Review by jfharden — APPROVED** [review]

**Pull request merged**